src/main/java/org/folio/circulation/domain/policy/Period.java

```diff
@@ -191,10 +195,20 @@ public int toMinutes() {
   }
 
   public boolean isLessThanOrEqualTo(Period otherPeriod) {
-    return toMinutes() <= otherPeriod.toMinutes();
+    return getMillisecondsDifference(otherPeriod) <= 0;
   }
 
   public boolean isGreaterThanOrEqualTo(Period otherPeriod) {
-    return toMinutes() >= otherPeriod.toMinutes();
+    return getMillisecondsDifference(otherPeriod) >= 0;
+  }
+
+  private long getMillisecondsDifference(Period other) {
+    // find ms difference by adding this period and other period
+    // to the same date and time point and subtracting milliseconds
+    final DateTime dateTimePoint = DateTime.now(DateTimeZone.UTC);
+    final DateTime thisDateTimeFromNowPoint = dateTimePoint.plus(timePeriod());
+    final DateTime otherDateTimeFromNowPoint = dateTimePoint.plus(other.timePeriod());
+
+    return thisDateTimeFromNowPoint.getMillis() - otherDateTimeFromNowPoint.getMillis();
   }
```

Comment:
We can easily do so for `minutes`, `hours`, `days` and `weeks` but it does not work for `months`, because they have vary duration.

Reply:
I think we need to be careful with how we define this method.

From an external perspective it is unclear that the difference is being determined relative to the current date and time.

Why is it relative to the current date and time? 

What is this method used for working out, whether one lost item policy period would exceed another?

Reply:
There is already a `toMinutes()` method in the class, which assumes that each class is 31 days, we can reuse it. But I think it is not accurate enough.

 > What is this method used for working out, whether one lost item policy period would exceed another?

Yes, it is used to determine whether _age to lost after period_ has passed since due date / or if refund interval has not passed since declaring an item lost.

Please also note, that this method is private, it is used in `isLess/GreaterThanOrEqual()` methods, which I believe does not affected by the using of some time point.

Reply:
> But I think it is not accurate enough.

Have you worked through an example of this with Holly to demonstrate the impact of the two approaches and confirm the expected behaviour?

> Yes, it is used to determine whether age to lost after period has passed since due date / or if refund interval has not passed since declaring an item lost.

If we are trying to determine whether a period has passed since a particular date, could that date be involved in that process?

That description suggests to me code something like (very much pseudo code):
`
val dueDate = loan.getDueDate();
val ageToLostPeriod = lostPolicy.getAgedToLostPeriod();
val now = Clock.now();

if (now > ageToLostPeriod.addTo(dueDate)) {
    // do something
}
`

Does that represent the process you are describing?

bq. Please also note, that this method is private, it is used in isLess/GreaterThanOrEqual() methods, which I believe does not affected by the using of some time point.

A concern I have with this approach is that it makes the methods unstable over time for the same input.

As you stated above, this calculation can differ depending upon the duration of the month. To me that means that the month inferred from the use of now can change the result of these methods. Is that the case?

Reply:
Good point, I've simplified the logic by adding 3 new methods:
1. `Period#hasPassedSinceDateTillNow(DateTime startDate)`:
```python
return startDate + this.timePeriod() is before now() or is now();
```
1. `Period#hasNotPassedSinceDateTillNow(DateTime startDate)`:
```java
return !hasPassedSinceDateTillNow(startDate)
```
1. `Period#isPassingNow(DateTime startDate)`:
```python
return startDate + this.timePeriod() is now()
```

Reply:
Cool, thanks, I like the direction those methods are heading in they are more expressive.

Personally, I think it might be easier to test if either a clock or date time is passed into the method, relying on a raw DateTime.now can cause challenges.

Reply:
I'll postpone this change, I think we can easily introduce this when needed. Is it ok?